lib/Sema/TypeCheckDeclPrimary.cpp

```diff
@@ -1279,6 +1279,7 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 public:
   ASTContext &Ctx;
   SourceFile *SF;
+  bool LeaveFunctionBodyUnchecked;
```

Comment:
Is there a way to reuse DeclChecker::shouldSkipBodyTypechecking so we don't have to have callers thread this around?

Reply:
If you wish, you could temporarily set the ASTContext in some kind of "code completion mode" if there isn't one already and read it there.

Reply:
Well, it turned out that `LeaveFunctionBodyUnchecked` is never `true` at least for now. I just removed the flag from `DeclChecker` so this file has no diffs anymore. Although there's still a flag in ConstraintSystem, I think it's better than a "global" flag in ASTContext.